script/setup/install-runc

```diff
@@ -27,19 +27,20 @@ function install_runc() {
 	cd "$GOPATH"/src/github.com/opencontainers/runc
 	git checkout $RUNC_COMMIT
 	make BUILDTAGS='apparmor seccomp selinux' runc
-	sudo make install
+	install=$(make install)
+	[ "$(id -u)" != "0" ] && sudo $install || $install
 }
 
 function install_crun() {
 	CRUN_VERSION=0.11
-	sudo curl -o /usr/local/sbin/runc -L https://github.com/containers/crun/releases/download/${CRUN_VERSION}/crun-${CRUN_VERSION}-static-$(uname -m)
-	sudo chmod +x /usr/local/sbin/runc
+	curl -o /usr/local/sbin/runc -L https://github.com/containers/crun/releases/download/${CRUN_VERSION}/crun-${CRUN_VERSION}-static-$(uname -m)
+	chmod +x /usr/local/sbin/runc
 }
 
 : ${RUNC_FLAVOR=runc}
 case ${RUNC_FLAVOR} in
 runc) install_runc ;;
-crun) install_crun ;;
+crun) [ "$(id -u)" != "0" ] && sudo bash -c "$(declare install_crun); install_crun" || install_crun ;;
```

Comment:
I think if you update https://github.com/containerd/containerd/blob/327c92f7d79a4ffb7972b3edd1b30bd372b41358/.github/workflows/ci.yml#L287-L288 to use `sudo`, that this would not be needed